content/lessons/chapter-3/coop-2.tsx

```diff
@@ -101,6 +106,18 @@ export default function Coop2({ lang }) {
     setAntagonistsBlockAmount(newBlock)
   }
 
+  function findIndexOfHighestHashpower(players, key) {
+    let highestScore = -Infinity
+    let indexOfHighestScore = 0
+    for (let i = 0; i < players.length; i++) {
+      if (players[i][key] > highestScore) {
+        highestScore = players[i][key]
+        indexOfHighestScore = i
+      }
+    }
+    return indexOfHighestScore
+  }
+
```

Comment:
If you are passing all the data this function needs via arguments we can move this function outside of the hook.
Also I think you can use a `.reduce` for this:
```ts
const indexOfHighestScore = players.reduce((highestIndex, current, index) => 
  current.value > players[highestIndex].value ? index : highestIndex,
0)
```